scrapyd/environ.py

```diff
@@ -30,11 +32,39 @@ def get_environment(self, message, slot):
         if project in self.settings:
             env['SCRAPY_SETTINGS_MODULE'] = self.settings[project]
         if self.logs_dir:
-            env['SCRAPY_LOG_FILE'] = self._get_file(message, self.logs_dir, 'log')
+            env['SCRAPY_LOG_FILE'] = self._get_log_file(message)
         if self.items_dir:
             env['SCRAPY_FEED_URI'] = self._get_feed_uri(message, 'jl')
         return env
 
+    def _get_log_file(self, message):
+        """Combine the logs_dir and logs_filename config items, and substitute
+        any variables in logs_filename to get the full filename of the log file
+        """
+        if not self.logs_filename:
+            # if no filename specified, fall back on the default.
+            return self._get_file(message, self.logs_dir, 'log')
+
+        now = datetime.datetime.now()
+        format_args = {}
+        format_args['project'] = message['_project']
+        format_args['spider'] = message['_spider']
+        format_args['Y'] = now.year
+        format_args['m'] = now.month
+        format_args['d'] = now.day
+        format_args['H'] = now.hour
+        format_args['M'] = now.minute
+        format_args['S'] = now.second
+        filename = self.logs_filename.format(**format_args)
+
+        full_filename = os.path.join(self.logs_dir, filename)
+
+        containing_dir = os.path.dirname(full_filename)
+        if not os.path.exists(containing_dir):
```

Comment:
We could test for it, but what should we do if we can't write to it?

I can't find any relevant Exception to throw. How would want this to be handled?

(In practice, trying to create the dir will raise an Exception if the dir isn't writable and the process dies.)